packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -240,43 +249,56 @@ class _DraggableSheetExtent {
   _DraggableSheetExtent({
     required this.minExtent,
     required this.maxExtent,
+    required this.snapPoints,
     required this.initialExtent,
     required VoidCallback listener,
-  }) : assert(minExtent != null),
-       assert(maxExtent != null),
-       assert(initialExtent != null),
-       assert(minExtent >= 0),
-       assert(maxExtent <= 1),
-       assert(minExtent <= initialExtent),
-       assert(initialExtent <= maxExtent),
-       _currentExtent = ValueNotifier<double>(initialExtent)..addListener(listener),
-       availablePixels = double.infinity;
+  })  : assert(minExtent != null),
+        assert(maxExtent != null),
+        assert(initialExtent != null),
+        assert(minExtent >= 0),
+        assert(maxExtent <= 1),
+        assert(minExtent <= initialExtent),
+        assert(initialExtent <= maxExtent),
+        _currentExtent = ValueNotifier<double>(initialExtent)
+          ..addListener(listener),
+        availablePixels = double.infinity;
 
   final double minExtent;
   final double maxExtent;
+  final Set<double> snapPoints;
   final double initialExtent;
   final ValueNotifier<double> _currentExtent;
   double availablePixels;
 
   bool get isAtMin => minExtent >= _currentExtent.value;
+
   bool get isAtMax => maxExtent <= _currentExtent.value;
 
   set currentExtent(double value) {
     assert(value != null);
     _currentExtent.value = value.clamp(minExtent, maxExtent);
   }
+
   double get currentExtent => _currentExtent.value;
 
   double get additionalMinExtent => isAtMin ? 0.0 : 1.0;
+
   double get additionalMaxExtent => isAtMax ? 0.0 : 1.0;
 
   /// The scroll position gets inputs in terms of pixels, but the extent is
   /// expected to be expressed as a number between 0..1.
-  void addPixelDelta(double delta, BuildContext context) {
+  void addPixelDelta(double pixelDelta, BuildContext context) {
     if (availablePixels == 0) {
       return;
     }
-    currentExtent += delta / availablePixels * maxExtent;
+    updateExtent(
+        currentExtent + pixelDelta / availablePixels * maxExtent, context);
+  }
+
+  /// Set the extent to the give new value. [newExtent] should be a number
+  /// between 0..1.
+  void updateExtent(double newExtent, BuildContext context) {
```

Comment:
Broke this out so that it could be called directly from the snapping animation.

The logic of this function could be achieved in a less error prone fashion (ie it'd be harder to accidentally change the extent without sending a notification) with a listener on `_currentExtent`-AFAIK there are no circumstances where we want to set the `_currentExtent.value` without also emitting a scroll notification. Is that something that would be reasonable to add to this PR?